mlir/lib/ExecutionEngine/ROCm/BackendUtils.cpp

```diff
@@ -39,19 +39,106 @@
 // TODO: remove this once the rocm_agent_enumerator is ready
 #include "hip/hip_runtime.h"
 
+#include <cstring>
 #include <mutex>
 #include <numeric>
 
 using namespace mlir;
 
 static constexpr const char kRunnerProgram[] = "mlir-rocm-runner";
 static constexpr const char kRocmAgentEnumerator[] = "rocm_agent_enumerator";
+static constexpr const char gcnArchDelimiter[] = ":";
+static constexpr const char targetTriple[] = "amdgcn-amd-amdhsa";
+
+namespace {
+// TODO: remove this once the rocm_agent_enumerator is ready
+void getGpuGCNArchName(hipDevice_t device, std::string &gcnArchName) {
+  hipDeviceProp_t props;
+  hipError_t result = hipGetDeviceProperties(&props, device);
+  if (result != hipSuccess) {
+    gcnArchName = "";
+    return;
```

Comment:
Is this just a `return;` situation? Do we instead want to crash? What does the state of the program look like after this return?

Reply:
It will error out later because the empty string arch is unexpected. I think it is better to crash early rather than later. What'd you suggest to do at this case?

Reply:
There's an argument for `llvm_fatal_error` here, but the actual solution is to go back and add validations at the place where the architecture is passed in from somewhere external and then to `llvm_unreachable()` out this case on the principle that it should already have been deteced. Or, if that's not too far up the call stack, we may want to pull out the `Error` or `Expected<>` classes https://llvm.org/docs/ProgrammersManual.html#recoverable-errors

Reply:
Responded below, I'd prefer a `llvm_unreachable()` on both cases.